### PR TITLE
Fix projection name

### PR DESCRIPTION
**Pull request by couet**
Fix https://root-forum.cern.ch/t/project3d-letter-d-in-name-option/57612

### CODE REVIEW

**Review by dpiparo — CHANGES_REQUESTED** [review]
Thanks @couet for this improvement. Before merging, could you add a test?

File: hist/hist/src/TH3.cxx [hunk 1]
```diff
@@ -2362,7 +2362,15 @@ TH2D *TH3::DoProject2D(const char* name, const char * title, const TAxis* projX,
 
 TH1 *TH3::Project3D(Option_t *option) const
 {
-   TString opt = option; opt.ToLower();
+   TString opt = option;
+   TString extra_name = option;
+   Int_t underscore = extra_name.Index("_");
```
Comment (lmoneta) [inline hist/hist/src/TH3.cxx:2367]: I think it is better to use the "Last" function to find the last accurence of "_". The user provided name could have an "_" 
while not the option string.

**Comment by couet** [comment]
@lmoneta @dpiparo, I applied your suggestions. Thanks!   
I guess it is ready to be approved.

File: hist/hist/src/TH3.cxx [hunk 2]
```diff
@@ -2410,7 +2418,11 @@ TH1 *TH3::Project3D(Option_t *option) const
 
    TString name = GetName();
    TString title = GetTitle();
-   name  += "_";   name  += opt;  // opt may include a user defined name
+   if (underscore>0) {
```
Comment (bellenot) [inline hist/hist/src/TH3.cxx:2421]:
```suggestion
   if (underscore > 0) {
```

File: hist/hist/src/TH3.cxx [hunk 3]
```diff
@@ -2362,7 +2362,15 @@ TH2D *TH3::DoProject2D(const char* name, const char * title, const TAxis* projX,
 
 TH1 *TH3::Project3D(Option_t *option) const
 {
-   TString opt = option; opt.ToLower();
+   TString opt = option;
+   TString extra_name = option;
+   Int_t underscore = extra_name.Last('_');
+   if (underscore>0) {
```
Comment (bellenot) [inline hist/hist/src/TH3.cxx:2368]:
```suggestion
   if (underscore > 0) {
```

**Review by bellenot — CHANGES_REQUESTED** [review]

File: hist/hist/test/test_Project3D_name.cxx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+#include "gmock/gmock.h"
+using ::testing::MatchesRegex;
+
+#include "TH2D.h"
+#include "TH3D.h"
+
+TEST(TH3, Project3D) {
+   auto h3 = new TH3D("h3","Title",10,0.,1.,10,0.,1.,10,0.,1.);
+   auto h2 = (TH2D*)h3->Project3D("Projection_xy");
+   EXPECT_THAT(h2->GetName() , MatchesRegex("h3_Projection_xy"));
```
Comment (dpiparo) [inline hist/hist/test/test_Project3D_name.cxx:13]: Does "MatchesRegex" mean that only the string "h3_Projection_xy" it it's matched? Identity should be verified in this case I think.

**Review by dpiparo — APPROVED** [review]

**Comment by dpiparo** [comment]
Yes I approved, with a small question about the test. Could you confirm it's checking for identity with the expected title string and not matching of some sort?

File: hist/hist/test/test_Project3D_name.cxx (hunk 4 above)
Comment (couet) [reply hist/hist/test/test_Project3D_name.cxx:13]: I'll check. That was the way I found to compare strings.

Comment (couet) [reply hist/hist/test/test_Project3D_name.cxx:13]: One also can use HasSubstr it seems ...

**Comment by couet** [comment]
> Yes I approved, with a small question about the test. Could you confirm it's checking for identity with the expected title string and not matching of some sort?

`StrEq` is the one. It has not been used until now in any other tests. That's why I did not find it immediately.

**Review by bellenot — APPROVED** [review]

**Comment by dpiparo** [comment]
This PR fixes https://github.com/root-project/root/issues/14331

**Pull request merged**